File: 2019/day1.py
import math

from util.runner import Runner


def calculate_fuel(mass: int):
    return math.floor(mass / 3) - 2

def calculate_total_fuel(mass: int):
    total = 0
    added = True

    while added:
        fuel = calculate_fuel(mass)
        if fuel > 0:
            total += fuel
            mass = fuel
        else:
            added = False

    return total

class Day1(Runner):
    def part1(self):
        total_fuel = 0
        for line in self.input:
            mass = int(line)
            fuel = calculate_fuel(mass)
            total_fuel += fuel
        return total_fuel

    def part2(self):
        total_fuel = 0
        for line in self.input:
            mass = int(line)
            fuel = calculate_total_fuel(mass)
            total_fuel += fuel
        return total_fuel


Day1(int)

File: 2025/day1/day1.go
package main

import (
	"2025/runner"
	"2025/types"
	"2025/util"
	"strconv"
)

func part1(input string) string {
	lines := util.AsLines(input)

	commands := parseLines(lines)

	point := 50
	total := 0

	for _, command := range commands {
		value, _ := strconv.Atoi(command[1])
		if command[0] == "R" {
			point += value
		}
		if command[0] == "L" {
			point -= value
		}
		if point%100 == 0 {
			total += 1
		}
	}

	return strconv.Itoa(total)
}

func part2(input string) string {
	lines := util.AsLines(input)

	commands := parseLines(lines)

	point := 50
	total := 0

	for _, command := range commands {
		value, _ := strconv.Atoi(command[1])

		fullTurns := value / 100
		partTurn := value % 100

		start := point

		total += fullTurns

		if command[0] == "R" {
			point += partTurn
		}
		if command[0] == "L" {
			point -= partTurn
		}
		if point == 0 {
			total += 1
		}
		if point >= 100 {
			if start != 0 {
				total += 1
			}
			point -= 100
		}
		if point < 0 {
			if start != 0 {
				total += 1
			}
			point += 100
		}
	}

	return strconv.Itoa(total)
}

func parseLines(lines []string) [][]string {
	var commands [][]string

	for _, line := range lines {
		command := []string{
			line[0:1],
			line[1:],
		}
		commands = append(commands, command)
	}

	return commands
}

func main() {
	day := types.Day{
		Part1: part1,
		Part2: part2,
		Day:   "1",
	}

	runner.RunDay(day)
}
